python/ray/air/checkpoint.py

```diff
@@ -351,6 +351,26 @@ def from_directory(cls, path: str) -> "Checkpoint":
         """
         return cls(local_path=path)
 
+    @classmethod
+    def from_checkpoint(cls, other: "Checkpoint") -> "Checkpoint":
```

Comment:
(Not for the PR) super nit: We should update the return types to match the Self type.